### DISCUSSION TITLE
Annoying animation on website element injected by extension (Multiselect for YouTube)

### DISCUSSION THREAD

**Discussion by mara004** [uBlockOrigin/uAssets#23423] [Q&A] [answered]
### Prerequisites

- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367), [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) or [_a shortener/hosting site_](https://github.com/uBlockOrigin/uAssets/discussions/17361) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Search [here](https://github.com/uBlockOrigin/uAssets/issues) to find out.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin along with other content blockers.
- [X] I have verified that the web browser's built-in content blocker/tracking protection, [network wide/DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking), or my VPN is not causing the issue.
- [X] I have verified that other extensions are not causing the issue.
- [X] If this is about a breakage or detection, I have verified that it is caused by uBlock Origin and isn't a site issue.
- [ ] I did not answer truthfully to **ALL** the above checkboxes.

### URL(s) where the issue occurs.

```yaml
https://www.youtube.com/ with Multiselect for YouTube Extension (Note: this is not about YouTube itself, but about the toolbar added into the website by the extension, so I believe this is not suitable for the dedicated YouTube thread)
```

### Description

The Multiselect for YouTube extension allows to select multiple videos and perform an action on the whole group (e.g. add to playlist). For this purpose, it seems to inject a button and toolbar into the website (toolbar appears when button is pressed).

On the toolbar, there is a quite annoying red heart animation beside a donation button. The animation serves no functional purpose, but is highly distracting to the user.
Since the toolbar appears to be part of the website, it should be possible to block the animation, but I am struggling to create a corresponding filter. The heart is not selectable using uBO's picker, but can be located through Firefox dev tools (animations tab).

The CSS Path is `html body div#msfy-bar-lvctqz07 div#content yt-icon#msfy-donate-icon-lvctqz07`, where `lvctqz07` seems to be some dynamic hash that changes each time the website is reloaded.
I tried to create a filter but could not find out how to match the dynamic part. (I found https://github.com/uBlock-LLC/uBlock/issues/1677 as related case, but it does not provide a solution.) Any help would be appreciated.

It's an odd case as the element comes from an extension, yet it seems to me like a candidate for the uBlock Annoyances list, assuming it is possible to create a filter. WDYT?

PS: I'm new to uAssets issues. While I acted according to my best knowledge, feel free to redirect me if this is nevertheless not the right place.

### Other extensions used

Multiselect for YouTube

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

Screen recording: [2024-04-23 22-48-33.zip](https://github.com/uBlockOrigin/uAssets/files/15082751/2024-04-23.22-48-33.zip) (see `.mkv` file in archive - not permitted for direct GH upload)

Snapshot:
![image](https://github.com/uBlockOrigin/uAssets/assets/65915611/74980565-be29-4e13-ae64-9cff6db0685d)

</details>

### Configuration

<details>
<summary>Details</summary>

```yaml
uBlock Origin: 1.57.2
Firefox: 120
filterset (summary):
 network: 190737
 cosmetic: 203673
 scriptlet: 49558
 html: 2036
listset (total-discarded, last-updated):
 added:
  adguard-generic: 81053-3929, 2h.11m
  adguard-spyware-url: 1501-120, 2h.11m
  block-lan: 61-0, 22d.23h.21m
  curben-phishing: 14197-1, 8h.30m
  fanboy-cookiemonster: 66950-16270, 8d.22h.25m
  ublock-annoyances: 6887-106, 2h.56m Δ
  dpollock-0: 11669-393, 2d.22h.4m
  DEU-0: 7408-41, 8h.30m
  adguard-cookies: 29308-3730, 2h.11m
  adguard-popup-overlays: 27152-328, 2h.11m
  adguard-mobile-app-banners: 5149-21, 2h.11m
  adguard-other-annoyances: 14167-357, 2h.11m
  [4 lists not shown]: [too many]
 default:
  user-filters: 4-0, never
  ublock-filters: 37758-109, 2h.56m Δ
  ublock-badware: 8384-0, 2h.56m Δ
  ublock-privacy: 833-1, 2h.56m Δ
  ublock-unbreak: 2316-2, 2h.56m Δ
  ublock-quick-fixes: 184-14, 2h.56m Δ
  easylist: 86302-1615, 2h.56m Δ
  easyprivacy: 50796-158, 2h.56m Δ
  urlhaus-1: 13565-0, 8h.30m
  plowe-0: 3736-1538, 2d.22h.4m
filterset (user): [array of 4 redacted]
trustedset:
 added: [array of 18 redacted]
userSettings:
 advancedUserEnabled: true
 showIconBadge: false
hiddenSettings: [none]
supportStats:
 allReadyAfter: 11680 ms (selfie)
 maxAssetCacheWait: 2316 ms
 cacheBackend: indexedDB
```
</details>

**Comment by mara004** [comment]
(Note that the toolbar is essential for the extension's functionality – the multiselect features can't be used unless the toolbar is exposed.)

**Comment by mara004** [comment, marked as answer]
Gotcha, with some further web search and try & error, I managed to write the filter:
`www.youtube.com##div[id^=msfy-bar] > div#content > yt-icon[id^=msfy-donate-icon]`

To hide the yellow donate button as well (though it's not strictly an annoyance):
`www.youtube.com##[id^=msfy-donate-button] > button`

*Edit: removed `-lvc` suffix as it also seems to by dynamic.*

**Comment by mara004** [comment]
(see also https://github.com/gorhill/uBlock/issues/619)

**Comment by stephenhawk8054** [comment]
DOM elements created by other extensions won't be addressed in uAssets. You can use the filters as custom filters. I'll convert this into discussion.
